src/utils/src/ExponentialBackoffUtils.c

```diff
@@ -0,0 +1,222 @@
+
+CleanUp:
+    LEAVES();
+    return retStatus;
+}
+
+STATUS resetExponentialBackoffRetryState(PExponentialBackoffState pExponentialBackoffState) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pExponentialBackoffState != NULL, STATUS_NULL_ARG);
+
+    pExponentialBackoffState->currentRetryCount = 0;
+    pExponentialBackoffState->lastRetryWaitTime = 0;
+    pExponentialBackoffState->status = BACKOFF_NOT_STARTED;
+    DLOGD("Resetting Exponential Backoff State");
+
+CleanUp:
+    LEAVES();
+    return retStatus;
+}
+
+STATUS exponentialBackoffStateWithDefaultConfigCreate(PExponentialBackoffState* ppExponentialBackoffState) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PExponentialBackoffState pExponentialBackoffState = NULL;
+    PExponentialBackoffConfig pExponentialBackoffConfig = NULL;
+
+    CHK(ppExponentialBackoffState != NULL, STATUS_NULL_ARG);
+
+    pExponentialBackoffState = (PExponentialBackoffState) MEMCALLOC(0x00, SIZEOF(ExponentialBackoffState));
+    CHK(pExponentialBackoffState != NULL, STATUS_NOT_ENOUGH_MEMORY);
+
+    CHK_STATUS(initializeDefaultExponentialBackoffConfig(&(pExponentialBackoffState->exponentialBackoffConfig)));
+    CHK_STATUS(resetExponentialBackoffRetryState(pExponentialBackoffState));
+
+CleanUp:
+    if (STATUS_SUCCEEDED(retStatus)) {
+        (*ppExponentialBackoffState) = pExponentialBackoffState;
+    } else {
+        SAFE_MEMFREE(pExponentialBackoffState);
+    }
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS inRange(UINT64 value, UINT64 low, UINT64 high) {
+    return value >= low && value <= high ? STATUS_SUCCESS : STATUS_INVALID_ARG;
+}
+
+STATUS validateExponentialBackoffConfig(PExponentialBackoffConfig pBackoffConfig) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pBackoffConfig != NULL, STATUS_NULL_ARG);
+
+    CHK_STATUS(inRange(pBackoffConfig->maxWaitTime, DEFAULT_KVS_MAX_WAIT_TIME_MILLISECONDS, KVS_BACKEND_STREAMING_IDLE_TIMEOUT_MILLISECONDS));
+    CHK_STATUS(inRange(pBackoffConfig->retryFactorTime, DEFAULT_KVS_RETRY_TIME_FACTOR_MILLISECONDS, LIMIT_KVS_RETRY_TIME_FACTOR_MILLISECONDS));
+    CHK_STATUS(inRange(pBackoffConfig->minTimeToResetRetryState, DEFAULT_KVS_MIN_TIME_TO_RESET_RETRY_STATE_MILLISECONDS, KVS_BACKEND_STREAMING_IDLE_TIMEOUT_MILLISECONDS));
+    CHK_STATUS(inRange(pBackoffConfig->jitterFactor, DEFAULT_KVS_JITTER_FACTOR_MILLISECONDS, LIMIT_KVS_JITTER_FACTOR_MILLISECONDS));
+
+CleanUp:
+    LEAVES();
+    return retStatus;
+}
+
+STATUS exponentialBackoffStateCreate(PExponentialBackoffState* ppBackoffState, PExponentialBackoffConfig pBackoffConfig) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PExponentialBackoffState pExponentialBackoffState = NULL;
+
+    CHK(ppBackoffState != NULL && pBackoffConfig != NULL, STATUS_NULL_ARG);
+    CHK_STATUS(validateExponentialBackoffConfig(pBackoffConfig));
+
+    pExponentialBackoffState = (PExponentialBackoffState) MEMCALLOC(0x00, SIZEOF(ExponentialBackoffState));
```

Comment:
Same here, first argument to MEMCALLOC needs to be a 1, its the number, see:

https://en.cppreference.com/w/c/memory/calloc